docs/figure_block.jl

```diff
@@ -5,18 +5,129 @@ Documenter.Selectors.order(::Type{FigureBlocks})  = 8.0 # like @example
 
 module MakieDocsHelpers
+    import ImageTransformations
+    import Makie
+    import FileIO
+
+    struct Png
+        bytes::Vector{UInt8}
+        size_px::Tuple{Int,Int}
+        id::String
+    end
+
+    struct PageInfo
+        path::String
+        title::String
+    end
+
+    FIGURES = Dict{PageInfo,Vector{Png}}()
     struct AsMIME{M<:MIME,V}
         mime::M
         value::V
     end
 
     Base.show(io::IO, m::MIME"image/svg+xml", a::AsMIME{MIME"image/svg+xml"}) = show(io,m, a.value)
     Base.show(io::IO, m::MIME"image/png", a::AsMIME{MIME"image/png"}) = show(io,m, a.value)
+
+    function register_figure!(page, pagetitle, id, figurelike)
+        vec = get!(Vector, FIGURES, PageInfo(page, pagetitle))
+        Makie.update_state_before_display!(figurelike)
+        scene = Makie.get_scene(figurelike)
+        size_px = scene.viewport[].widths
+        img = Makie.colorbuffer(scene)
+        factor = round.(Int, size(img) ./ size_px)
+        ntrim = 3
+        img = @view ImageTransformations.restrict(img)[ntrim:end-ntrim,ntrim:end-ntrim]
+        # img = @view ImageTransformations.restrict(img)[ntrim:end-ntrim,ntrim:end-ntrim]
+        final_size_px = Tuple(round.(Int, size(img) ./ factor))
+        io = IOBuffer()
+        FileIO.save(FileIO.Stream{FileIO.format"PNG"}(Makie.raw_io(io)), img)
+        push!(vec, Png(take!(io), final_size_px, id))
+        return
+    end
+
+    struct FileInfo
+        filename::String
+        id::String
+        size_px::Tuple{Int,Int}
+    end
+    struct OverviewSection
+        d::Dict{PageInfo,Vector{FileInfo}}
+    end
+
+    function OverviewSection(page::String)
+        r = Regex("/$page/")
+
+        filtered = filter(pairs(FIGURES)) do (pageinfo, pngs)
+            match(r, pageinfo.path) !== nothing
+        end
+
+        fileinfo_dict = Dict{PageInfo,Vector{FileInfo}}()
+        for (pageinfo, pngs) in pairs(filtered)
+            fileinfos = map(pngs) do png
+                filename = "$(string(hash(png.bytes), base = 62)).png"
+                open(filename, "w") do _io
+                    write(_io, png.bytes)
+                end
+                return FileInfo(filename, png.id, png.size_px)
+            end
+            fileinfo_dict[pageinfo] = fileinfos
+        end
+
+        OverviewSection(fileinfo_dict)
+    end
+
+    function Base.show(io::IO, ::MIME"text/markdown", o::OverviewSection)
+        pages = sort(collect(keys(o.d)), by = x -> x.path)
+        for page in pages
+            fileinfos = o.d[page]
+            pagename, _ = splitext(basename(page.path))
+            println(io, "### [$(page.title)](@id overview_$(pagename))")
+            println(io)
+            println(io, """<div :class="\$style.container">""")
+            for fileinfo in fileinfos
+                println(io, """
+                <a href="./$pagename.html#example-$(fileinfo.id)">
+                    <img src=\"./$(fileinfo.filename)\" />
```

Comment:
I'm not sure this will work with Vitepress because of rollup - may want to switch this to including the PNG base64 directly.